source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -168,8 +147,17 @@ public void IdentifyAsServer()
 
         public RemoteIdentity ReadRemoteIdentity()
         {
-            var line = streamReader.ReadLine();
+            var line = controlMessagesReader.ReadControlMessage(stream);
+            
+            
             if (string.IsNullOrEmpty(line)) throw new ProtocolException("Unable to receive the remote identity; the identity line was empty.");
+            
+            var emptyLine = controlMessagesReader.ReadControlMessage(stream);
```

Comment:
Explicitly read the empty line that is sent after the identity line, rather than rely on the `StreamReader` to happen to consume that within its buffer.